WordPress/Sniffs/Theme/DeregisterScriptSniff.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Sniff to check theme for wp_deregister_style()
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   Simon Prosser <pross@pross.org.uk>
+ */
+class WordPress_Sniffs_Theme_DeregisterScriptSniff extends WordPress_Sniffs_Functions_FunctionRestrictionsSniff
+{
+
+	/**
+	 * Exclude groups
+	 *
+	 * Example: 'switch_to_blog,user_meta'
+	 *
+	 * @var string Comma-delimited group list
+	 */
+	public $exclude = '';
+
+	/**
+	 * Groups of functions to restrict
+	 *
+	 * Example: groups => array(
+	 * 	'lambda' => array(
+	 * 		'type' => 'error' | 'warning',
+	 * 		'message' => 'Use anonymous functions instead please!',
+	 * 		'functions' => array( 'eval', 'create_function' ),
+	 * 	)
+	 * )
+	 *
+	 * @return array
+	 */
+	public function getGroups() {
+		return array(
+			'wp_deregister_script' => array(
+				'type'	=> 'error',
```

Comment:
The current error level is "warning" as there is no check against a list of core registered scripts and the rule - as it is in the handbook - only applies to those.
If you think the level should change to "error", please ask the Theme Review Board for approval.

In that case, I would have to say that that also implies that a list with core scripts which are not allowed to be deregistered, should be added, maintained and checked against.

If you think the rule should be an error for deregistering of any script, not just core scripts, again, please ask the Theme Review Board for approval.